packages/js/src/ui/components/elements/Bell/Bell.tsx

```diff
@@ -1,24 +1,32 @@
 import { createSignal, JSX, onCleanup, onMount } from 'solid-js';
+import { requestLock } from '../../../helpers/browser';
+import { NV_INBOX_WEBSOCKET_LOCK } from '../../../helpers/constants';
 import { useNovu } from '../../../context';
 import { BellContainer } from './DefaultBellContainer';
+import { useTabsChannel } from '../../../helpers/useTabsChannel';
 
 type BellProps = {
   children?: ({ unreadCount }: { unreadCount: number }) => JSX.Element;
 };
 export function Bell(props: BellProps) {
   const [unreadCount, setUnreadCount] = createSignal(0);
   const novu = useNovu();
+  const { postMessage } = useTabsChannel<number>({ onMessage: setUnreadCount });
 
   const updateReadCount = (data: { result: number }) => {
     setUnreadCount(data.result);
+    postMessage(data.result);
   };
 
   onMount(() => {
-    novu.on('notifications.unread_count_changed', updateReadCount);
-  });
+    const resolveLock = requestLock(NV_INBOX_WEBSOCKET_LOCK, () => {
+      novu.on('notifications.unread_count_changed', updateReadCount);
```

Comment:
The WebSocket connection is established only when starting to listen on the one of WebSocket events. That's why wrapping it with Web Lock API is safe.